tracer/src/Datadog.Trace.Tools.Runner/CheckIisCommand.cs

```diff
@@ -0,0 +1,147 @@
+
+        public override ValidationResult Validate(CommandContext context, CheckIisSettings settings)
+        {
+            var result = base.Validate(context, settings);
+
+            if (result.Successful)
+            {
+                // Perform additional validation
+                if (settings.SiteName.Count(c => c == '/') > 1)
+                {
+                    return ValidationResult.Error($"IIS site names can't have multiple / in their name: {settings.SiteName}");
+                }
+            }
+
+            return result;
+        }
+
+        internal static async Task<int> ExecuteAsync(CheckIisSettings settings, string applicationHostConfigurationPath, int? pid)
+        {
+            var values = settings.SiteName.Split('/');
+
+            var siteName = values[0];
+            var applicationName = values.Length > 1 ? $"/{values[1]}" : "/";
+
+            AnsiConsole.WriteLine(FetchingApplication(siteName, applicationName));
+
+            var serverManager = new ServerManager(readOnly: true, applicationHostConfigurationPath);
+
+            var site = serverManager.Sites[siteName];
+
+            if (site == null)
+            {
+                Utils.WriteError(CouldNotFindSite(siteName, serverManager.Sites.Select(s => s.Name)));
+
+                return 1;
+            }
+
+            var application = site.Applications[applicationName];
+
+            if (application == null)
+            {
+                Utils.WriteError(CouldNotFindApplication(siteName, applicationName, site.Applications.Select(a => a.Path)));
+
+                return 1;
+            }
+
+            var pool = serverManager.ApplicationPools[application.ApplicationPoolName];
+
+            // The WorkerProcess part of ServerManager doesn't seem to be compatible with IISExpress
+            // so we skip this bit when launched from the tests
+            if (pid == null)
+            {
+                var workerProcesses = pool.WorkerProcesses;
+
+                if (workerProcesses.Count > 0)
+                {
+                    // If there are multiple worker processes, we just take the first one
+                    // In theory, all worker processes have the same configuration
+                    pid = workerProcesses[0].ProcessId;
+                }
+            }
+
+            if (pid == null)
+            {
+                Utils.WriteWarning(NoWorkerProcess);
+            }
+            else
+            {
+                AnsiConsole.WriteLine(InspectingWorkerProcess(pid.Value));
+
+                var rootDirectory = application.VirtualDirectories.FirstOrDefault(d => d.Path == "/")?.PhysicalPath;
+
+                IConfigurationSource appSettingsConfigurationSource = null;
+
+                try
+                {
+                    var config = application.GetWebConfiguration();
+                    var appSettings = config.GetSection("appSettings");
+                    var collection = appSettings.GetCollection();
```

Comment:
Just confirming - do any of these throw if web.config doesn't exist, or it doesn't have an appsettings section? I don't know if we would definitely want to log a warning in that case (though maybe an information message I guess, so I guess it doesn't matter!)

Reply:
There's always a config of some kind because it inherits the default IIS config. As far as my testing goes, it only throws if the web.config is invalid (in which case the web application doesn't work anyway).